### PR TITLE
[Backport 2.28] Update `SSL_SERIALIZED_SESSION_CONFIG_BITFLAG` with new flags

### PR DESCRIPTION
**Pull request by davidhorstmann-arm**
Simple but not trivial backport of #8821, created manually rather than through cherry-picking.

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [ ] **changelog** provided, or not required
- [x] **backport** of #8821 
- [x] **tests** not required - testing will be provided in a followup

## Notes for the submitter

Please refer to the [contributing guidelines](https://github.com/Mbed-TLS/mbedtls/blob/development/CONTRIBUTING.md), especially the
checklist for PR contributors.

Help make review efficient:
* Multiple simple commits
  - please structure your PR into a series of small commits, each of which does one thing
* Avoid force-push
  - please do not force-push to update your PR - just add new commit(s)
* See our [Guidelines for Contributors](https://mbed-tls.readthedocs.io/en/latest/reviews/review-for-contributors/) for more details about the review process.

### CODE REVIEW

**Comment by mpg** [comment]
> * **changelog** provided, or not required

I'd say required, since it's fixing a bug that was present in a released version.

File: library/ssl_tls.c [hunk 1]
```diff
@@ -5252,7 +5259,8 @@ const mbedtls_ssl_session *mbedtls_ssl_get_session_pointer(const mbedtls_ssl_con
          (SSL_SERIALIZED_SESSION_CONFIG_TRUNC_HMAC << \
              SSL_SERIALIZED_SESSION_CONFIG_TRUNC_HMAC_BIT) | \
          (SSL_SERIALIZED_SESSION_CONFIG_ETM << SSL_SERIALIZED_SESSION_CONFIG_ETM_BIT) | \
-         (SSL_SERIALIZED_SESSION_CONFIG_TICKET << SSL_SERIALIZED_SESSION_CONFIG_TICKET_BIT)))
+         (SSL_SERIALIZED_SESSION_CONFIG_TICKET << SSL_SERIALIZED_SESSION_CONFIG_TICKET_BIT) | \
+         (SSL_SERIALIZED_SESSION_KEEP_PEER_CRT << SSL_SERIALIZED_SESSION_CONFIG_KEEP_CRT_BIT)))
```
Comment (tom-cosgrove-arm) [inline library/ssl_tls.c:5263]: Same naming nit as in the development PR: all the others have `CONFIG` in the name, and just add `_BIT` to specify the bit. This one doesn't have `CONFIG` in the name (but does in the `_BIT` value, but the `_BIT` value removes `_PEER`.

While this was probably done for length reasons, there is `SSL_SERIALIZED_SESSION_CONFIG_CLIENT_TICKET_BIT` which is as long as `SSL_SERIALIZED_SESSION_CONFIG_KEEP_PEER_CRT_BIT` would be

**Review by tom-cosgrove-arm — DISMISSED** [review]
LGTM

**Comment by davidhorstmann-arm** [comment]
Feedback should be addressed. Most of the refactoring work doesn't apply to 2.28 as all of the session serialization code is already in one place.

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

File: library/ssl_tls.c [hunk 2]
```diff
@@ -5278,19 +5287,34 @@ static const unsigned char ssl_serialized_session_header[] = {
...
+ * #if defined(MBEDTLS_HAVE_TIME)
+ *     uint64 start_time;
+ * #endif
+ *     uint8 ciphersuite[2];        // defined by the standard
+ *     uint8 compression;           // 0 or 1
+ *     uint8 session_id_len;        // at most 32
+ *     opaque session_id[32];
+ *     opaque master[48];           // fixed length in the standard
+ *     uint32 verify_result;
+ * #if defined(MBEDTLS_SSL_KEEP_PEER_CERTIFICATE)
```
Comment (ronald-cron-arm) [inline library/ssl_tls.c:5299]:
```suggestion
 * #if defined(MBEDTLS_X509_CRT_PARSE_C)
 * #if defined(MBEDTLS_SSL_KEEP_PEER_CERTIFICATE)
```

File: library/ssl_tls.c [hunk 3]
```diff
@@ -5278,19 +5287,34 @@ static const unsigned char ssl_serialized_session_header[] = {
...
+ *     opaque peer_cert<0..2^24-1>; // length 0 means no peer cert
+ * #else
+ *     uint8 peer_cert_digest_type;
+ *     opaque peer_cert_digest<0..2^8-1>
+ * #endif
+ * #if defined(MBEDTLS_SSL_SESSION_TICKETS)
```
Comment (ronald-cron-arm) [inline library/ssl_tls.c:5305]:
```suggestion
 * #if defined(MBEDTLS_SSL_SESSION_TICKETS) && defined(MBEDTLS_SSL_CLI_C)
```

**Review by ronald-cron-arm — CHANGES_REQUESTED** [review]
Just two guards that are missing in the comments I think.

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Review by ronald-cron-arm — APPROVED** [review]
Thanks for addressing my comments. LGTM. The CI is in a strange state though.

**Comment by tom-cosgrove-arm** [comment]
The RTD build failure looks like a transient issue. I've re-started the CI on OpenCI (but that may not be the right thing to have done)

**Comment by davidhorstmann-arm** [comment]
ReadTheDocs is separate, but I've restarted it

**Comment by gilles-peskine-arm** [comment]
Merging is blocked because the DCO check didn't report back. I'll try the close-reopen trick.

**Comment by gilles-peskine-arm** [comment]
Close-reopen did not work to re-trigger the DCO check. I'm out of ideas other than pushing a new commit (with an updated time or whatever).

**Comment by davidhorstmann-arm** [comment]
(Rebasing the last commit with no changes seems to have worked)

**Pull request merged**